tests/functions/test_fill_empty.py

```diff
@@ -11,3 +12,22 @@ def test_fill_empty(null_df):
 def test_fill_empty_column_string(null_df):
     df = null_df.fill_empty(column_names="2", value=3)
     assert set(df.loc[:, "2"]) == set([3])
+
+
+@pytest.mark.functions
+@pytest.mark.parametrize(
+    "column_names",
+    [
+        (0, 1, "2", "3"),  # tuple
+        [0, 1, "2", "3"],  # list
+        {0, 1, "2", "3"},  # set
+        ({0: 0, 1: 1, "2": "2", "3": "3"}).keys(),  # dict key
+        ({0: 0, 1: 1, "2": "2", "3": "3"}).values(),  # key value
+        pd.Index([0, 1, "2", "3"]),  # Index
+    ],
```

Comment:
`map(lambda x: x if x < 2 else str(x), range(4)),  # generator`

The generator is also `Iterable`, but this would work.

Reply:
That's a valuable thing to note. @Zeroto521 does the function raise an error when a generator is passed in? If so, then we may want to have a test for that case where we explicitly ask purest to expect an error when passing in a generator.

Reply:
https://github.com/pyjanitor-devs/pyjanitor/blob/d3146bbf0dddcb0cd979691bd171be69e12c0af5/janitor/functions/fill.py#L144-L145

`map(lambda x: x if x < 2 else str(x), range(4))` is `map` type and it can only use once.
The `check_column` consumes `column_names` data.

Reply:
One solution.

Let `check_column` return a new `list` type `column_names`.

https://github.com/pyjanitor-devs/pyjanitor/blob/d3146bbf0dddcb0cd979691bd171be69e12c0af5/janitor/utils.py#L298-L336

Reply:
Another solution.

Copy `column_names`

```python
from copy import deepcopy

 check_column(df, deepcopy(column_names)) 
 return _fill_empty(df, column_names, value=value)
```

Reply:
@thatlittleboy @samukweku what do you all think? I think I'd be in favour of the `deepcopy` implementation over re-casting to list; something feels wonky to do change of types, but I might be over-sensitive here.

Reply:
Deep-copying a generator doesn't seem to work (error is thrown about not being able to pickle a generator).
The canonical way is to convert the generator into a list to persist it, so I think I'll be in favour of that. But I suggest we leave this to a separate PR (changing the "api" of `check_columns` to take care of generators properly).

Reply:
There is a new issue #1015 to trace this.